content/learning-paths/cross-platform/integer-vs-floats/integer-float-conversions.md

```diff
@@ -0,0 +1,196 @@
+
+```
+gaussian:
+        stp     x29, x30, [sp, -32]!
+        mov     x29, sp
+        str     d8, [sp, 16]
+        fmov    d8, d0
+        fsub    d0, d2, d1
+        fadd    d1, d8, d8
+        fnmul   d0, d0, d0
+        fmul    d1, d1, d8
+        fdiv    d0, d0, d1
+        bl      exp
+        adrp    x0, .LC0
+        ldr     d1, [x0, #:lo12:.LC0]
+        fmul    d8, d8, d1
+        fmov    d1, 1.0e+0
+        fdiv    d1, d1, d8
+        ldr     d8, [sp, 16]
+        ldp     x29, x30, [sp], 32
+        fmul    d0, d1, d0
+        ret
+vpx_setup_noise:
+        stp     x29, x30, [sp, -336]!
+        mov     x29, sp
+        stp     x21, x22, [sp, 32]
+        mov     x22, x0
+        adrp    x0, .LC1
+        stp     d8, d9, [sp, 48]
+        fmov    d8, d0
+        mov     w21, w1
+        ldr     d9, [x0, #:lo12:.LC1]
+        stp     x19, x20, [sp, 16]
+        mov     w20, -32
+        mov     w19, 0
+        str     d10, [sp, 64]
+        fmov    d10, 5.0e-1
+.L9:
+        scvtf   d2, w20
+        movi    d1, #0
+        fmov    d0, d8
+        bl      gaussian
+        fmadd   d0, d0, d9, d10
+        fcvtzs  w1, d0
+        cbz     w1, .L5
+        add     x0, sp, 80
+        mov     x2, 0
+        add     x3, x0, w19, sxtw
+        b       .L4
+...
+```
+
+You can see the full code and its dissassembly in the [godbolt full link](https://godbolt.org/z/cTM8d3dq5))
+
+Notice that before and after the call to `gaussian`: `bl gaussian`, two conversion instructions are called: 
+
+```
+scvtf   d2, w20
+```
+and
+
+```
+fcvtzs w1, d0
+```
+
+which do exactly that, convert the value between register `d0` (a double register) from/to an integer value in registers `w1`, `w20`.
+
+Now, in this particular case this conversion is done on purpose and controlled but when dealing with performance critical software, such conversions can be costly and should be avoided unless there is no other alternative.
+But the situation at least is clear with explicit conversions, implicit ones are a bit harder to track.
+
+## Implicit Conversions
+
+When a conversion is not explicitly stated by the user, then it's an implicit one. In general it's a conversion that the compiler issues when there is an operation involving two (or more) elements of different datatypes.
+In that case, the compiler has to convert one of the values to the same datatype as the other, as most of the operations involve elements of the same size -with some exceptions, like for example the `SADDW`/`UADDW` Neon instructions which add elements of different widths. Such instructions do not require any kind of conversion. But let's consider the generic operation:
```

Comment:
I'll leave this up to techcomms, but generally speaking we refer to `Neon` in documentation as `Advanced SIMD`